Saule/Http/PaginatedAttribute.cs

```diff
@@ -34,23 +36,55 @@ public int PerPage
                     throw new ArgumentOutOfRangeException(nameof(PerPage), value, "Must have at least one item per page.");
                 }
 
+                if (_queryPageSizeLimit.HasValue && value > _queryPageSizeLimit)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(PageSizeLimit), value, "Items per page cannot be larger than the page size limit.");
+                }
+
                 _perPage = value;
             }
         }
 
         /// <summary>
+        /// Gets or sets the maximum page size to accept from a URL query string.
+        /// </summary>
+        public int PageSizeLimit
```

Comment:
Perhaps this should be `int?` instead of using `-1` to represent unlimited size? I'm curious about the reason you went for `int` here, but `int?` in the config.